internal/mode/static/state/graph/graph_test.go

```diff
@@ -360,3 +387,136 @@ func TestBuildGraph(t *testing.T) {
 		})
 	}
 }
+
+func TestIsReferenced(t *testing.T) {
+	baseSecret := &v1.Secret{
```

Comment:
Ok that's good to know, I forgot that I wasn't using BuildGraph and instead just creating a Graph object myself. If I were to use BuildGraph I know it would do some verification or such.